### PR TITLE
Switch from vmap to scan for coilset compute

### PR DESCRIPTION
**Pull request by f0uriest**
Saves a lot of memory and works better for heterogeneous coil types

### CODE REVIEW

File: desc/backend.py [hunk 1]
```diff
@@ -555,6 +556,45 @@ def fun_vmap(fun_inputs):
 
         return fun_vmap
 
+    def scan(f, init, xs, length=None, reverse=False, unroll=1):
```
Comment (rahulgaur104) [inline desc/backend.py:559]: I am assuming this is probably not vectorized.
So if we want to integrate an array of shape (L, M, N) along the last axis, will this work?

Comment (f0uriest) [reply desc/backend.py:559]: You'd need to transpose the array. This function is basically just a non-jax version of a jax function (for legacy reasons we've been trying to keep things compatible with pure numpy), so this is just trying to match the API of the jax version. In practice this never gets used, only the jax version `jax.lax.scan` (imported above)

**Review by dpanici — APPROVED** [review]

**Review by rahulgaur104 — APPROVED** [review]

File: desc/backend.py (hunk 1 above)
Comment (ddudt) [reply desc/backend.py:559]: Haven't we discussed not supporting regular numpy anymore? I would be in favor of that in for a new release

**Review by ddudt — APPROVED** [review]
